packages/core/src/document/editor.tsx

```diff
@@ -117,6 +88,22 @@ export const DocumentEditor: React.FunctionComponent<
     return null
   }
 
+  let state: unknown
+  if (isStatefulPlugin<StateDescriptor<any>>(plugin)) {
+    const onChange = (
+      updater: (value: unknown, helpers: StoreDeserializeHelpers) => void
+    ) => {
+      const updated = updater(document.state, deserializeHelpers)
```

Comment:
We should give the updater to the store instead. Otherwise, we might run into race conditions again.